registry/config.go

```diff
@@ -32,7 +32,7 @@ const (
 var (
 	ErrInvalidRepositoryName = errors.New("Invalid repository name (ex: \"registry.domain.tld/myrepos\")")
 	emptyServiceConfig       = NewServiceConfig(nil)
-	validNamespaceChars      = regexp.MustCompile(`^([a-z0-9-_]*)$`)
+	validNamespaceChars      = regexp.MustCompile(`^([a-z0-9-_/]*)$`)
```

Comment:
FYI, this is what the v2 registry currently does:

- component regex: https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L38
- full name regex: https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L46